src/saveload/afterload.cpp

```diff
@@ -2312,6 +2312,14 @@ bool AfterLoadGame()
 	if (IsSavegameVersionBefore(SLV_128)) {
 		const Depot *d;
 		FOR_ALL_DEPOTS(d) {
+			/* At some point, invalid depots were saved into the game (possibly those removed in the past?)
+			 * Remove them here, so they don't cause issues further down the line */
+			if (!IsDepotTile(d->xy)) {
+				DEBUG(sl, 0, "Removing invalid depot %d at %d, %d", d->index, TileX(d->xy), TileY(d->xy));
+				delete d;
+				d = nullptr;
```

Comment:
Why bother with this line?

Reply:
I did because (nearly) all the other pool deletions seem to do the same thing, presuambly due to
https://github.com/OpenTTD/OpenTTD/blob/f354b869c152652706e6eb16af7d14b07b06ef68/src/core/pool_type.hpp#L316-L318

Reply:
That explicitly nulls it every iteration as well... but if every other loop does it then what do I know.